infrastructure/smart-contracts/scripts/flagged/flaggedNodeLicenses.mjs

```diff
@@ -0,0 +1,50 @@
+import hardhat from "hardhat";
+import fs from 'fs';
+import { parse } from "csv/sync";
+import inquirer from 'inquirer';
+import { NodeLicenseAbi, config } from "@sentry/core";
+const { ethers } = hardhat;
+
+async function main() {
+    //prompt user for the path to the list
+    const { pathToList } = await inquirer.prompt({
+        type: 'input',
+        name: 'pathToList',
+        message: 'Enter the absolute path to the csv containing addresses to check:',
+    });
+
+    //validate csv path
+    if (!fs.existsSync(pathToList)) {
+        console.log("Invalid file path, file does not exists", pathToList);
+        return;
+    }
+
+    //read addresses from csv
+    const addressesToCheck = parse(fs.readFileSync(pathToList), { columns: true });
+
+    //connect to arbitrum one
+    const provider = new ethers.JsonRpcProvider(config.arbitrumOneJsonRpcUrl);
+
+    console.log("Checking balances of flagged wallets...");
+    let flaggedCount = 0
+    for (let i = 0; i < addressesToCheck.length; i++) {
+        //load contract with provider
+        const nodeLicense = await new ethers.Contract(config.nodeLicenseAddress, NodeLicenseAbi, provider);
```

Comment:
I did not notice this before, the contract init should happen outside the loop.

Reply:
Done